index.js

```diff
@@ -167,10 +167,18 @@ const validateResponse = (req, res, next) => {
         try {
           val = JSON.parse(val);
         } catch (err) {
+          res.set('Content-Type', ''); // Reset content-type since it is no longer valid
```

Comment:
It is necessary for this type of errors. If parsing failed due to response not being a proper json, chances are it was just text. Which means that content-type is likely to be set to text and error will not be returned correctly as we are trying to return json in body, but body will be ignored.

Reply:
I'm a bit on a fence with this one still. The library is meant to validate incoming/outgoing data, but modifying headers seems too much. If somebody is passing in an invalid JSON, the lib should definitely signal an error but the type of header the server is sending is up to the consumer to decide.
 
For example, if I were to use this library for the first time without looking into the internals, would never expect it to modify the headers for me because it should only deal with data validation :)

Reply:
But should we then be checking previously set Content-Type and adapt our response accordingly? So if user set response to be text, we return error as a text as well instead of a JSON as we typically do? That would make error handling on user's side complicated, he cannot simply expect all his error responses to follow consistent format, but check for both possible formats if he wants to react to error happening.
Note that we are not modifying headers in happy-path, only if validation fails, so response is already not what user is expecting to get in successful scenario.

Reply:
I can make this behaviour configurable with an option flag if you want, but I still believe that the most intuitive and helpful default would be enable enforcing error responses always to be of JSON format so that users can have consistent error handling on their side.

Reply:
Until there is a more in-depth discussion on this, I will preserve original behaviour even though I strongly believe it to be wrong, adjust tests accordingly and implement alternate handling behind the flag.